### PR TITLE
Bind to random free local port

### PR DESCRIPTION
**Pull request by jsuresh**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -218,8 +219,25 @@ func prepend(item interface{}, items []interface{}) []interface{} {
 	return append([]interface{}{item}, items...)
 }
 
+func getFreeLocalPort() (port int) {
+	l, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		log.Fatalf("Couldn't get free local port (to setup ssh tunnel)", err)
+	}
+	defer l.Close()
```
Comment (soulplant) [inline client.go]: No need to defer.

Comment (jsuresh) [inline client.go]: Don't understand? Just close it right away?

Comment (soulplant) [inline client.go]: Yep, sorry, that's what I meant.

**Comment by soulplant** [comment]
LGTM

File: client.go (hunk 1 above)
Comment (jsuresh) [inline client.go]: done.

**Pull request merged**